rust/src/router/webrtc_transport/tests.rs

```diff
@@ -124,18 +130,13 @@ fn create_with_webrtc_server_succeeds() {
 
         {
             let ice_candidates = transport.ice_candidates();
-            assert_eq!(ice_candidates.len(), 2);
+            assert_eq!(ice_candidates.len(), 1);
             assert_eq!(ice_candidates[0].ip, IpAddr::V4(Ipv4Addr::LOCALHOST));
-            assert_eq!(ice_candidates[0].protocol, Protocol::Udp);
-            assert_eq!(ice_candidates[0].port, port1);
+            assert_eq!(ice_candidates[0].protocol, Protocol::Tcp);
+            assert_eq!(ice_candidates[0].port, port2);
             assert_eq!(ice_candidates[0].r#type, IceCandidateType::Host);
```

Comment:
We had UDP and TCP before, why removing UDP?

Reply:
Because I am testing `enable_udp: false` as I said. Just a few lines above:

```rust
let transport = router
            .create_webrtc_transport({
                let mut webrtc_transport_options = WebRtcTransportOptions::new_with_server(
                    webrtc_server.clone()
                );
                // Let's disable UDP so resulting ICE candidates should only contain TCP.
                webrtc_transport_options.enable_udp = false;

                webrtc_transport_options
            })
            .await
            .expect("Failed to create WebRTC transport");
```

Reply:
Why is TCP testing no longer necessary here then?

Reply:
- That test creates a `WebRtcServer` that listens in UDP and TCP.
- Then it creates a transport with `webrtc_server` and `enable_udp: false`, meaning that the resulting transport should only use the TCP ICE candidates of the given `webrtc_server`.
- So the test asserts that there is only a ICE candidate (`assert_eq!(ice_candidates.len(), 1);`)  and it's TCP.

Reply:
I understand what test is doing with these changes, but I don't understand what was wrong with the old test that you needed to change it.

Should you maybe add another test case instead? Also why no such change in TS, did TS test contain different conditions? (tests are usually symmetrical).

Reply:
Fair. I've done same change in Node test: https://github.com/versatica/mediasoup/pull/1185/commits/cf0b0332438f73716ace87cc06ad6eb65db9668a

> Should you maybe add another test case instead?

There is little value in just asserting that a transport created with `webRtcServer` has the same ICE candidates than `webRtcServer`. So IMHO in those tests we can also test those `enable_udp` etc fields.

Reply:
Because:

- Jose and me have lot of work these days. I'm exhausted working on 3 different mediasoup PRs in parallel and doing things at work.
- The current API sucks. It got way worse after adding `webRtcServer` since then we allow `listenIps` OR `webRtcServer` but not both, and some other options only make sense for one of them, etc etc. I know it and I'd love to refactor it in v4, but we have to keep backwards compatibility.

But all those are excuses and you are right.